bundle/Core/FieldType/Tags/Type.php

```diff
@@ -114,7 +114,7 @@ public function fromHash($hash): Value
                         'parentTagId' => $hashItem['parent_id'],
                         'keywords' => $hashItem['keywords'],
                         'mainLanguageCode' => $hashItem['main_language_code'],
-                        'remoteId' => $hashItem['remote_id'] ?? null,
+                        'remoteId' => $hashItem['remote_id'] ?? md5(uniqid(static::class, true)),
```

Comment:
This should rather be an empty string than a generated remote ID. An empty string signals to the system non-existence of the remote ID.

Reply:
Ok, 

The probleme is on variable type declaration. Better solution ?
````
--- a/bundle/API/Repository/Values/Tags/Tag.php
+++ b/bundle/API/Repository/Values/Tags/Tag.php
@@ -78,7 +78,7 @@ final class Tag extends ValueObject
     /**
      * A global unique ID of the tag.
      */
-    protected string $remoteId;
+    protected ?string $remoteId;
 
````

Reply:
I think you misunderstood me :)

I was thinking more about

```
'remoteId' => $hashItem['remote_id'] ?? '',
```

This way you don't have to modify the remoteId property in Tag object.

Reply:
Hello,

I'm ok to not change this `'remoteId' => $hashItem['remote_id'] ?? '',`
the error is : 
````
Cannot assign null to property Netgen\TagsBundle\API\Repository\Values\Tags\Tag::$remoteId of type string

TypeError:
Cannot assign null to property Netgen\TagsBundle\API\Repository\Values\Tags\Tag::$remoteId of type string

  at vendor/ibexa/core/src/contracts/Repository/Values/ValueObject.php:33
  at Ibexa\Contracts\Core\Repository\Values\ValueObject->__construct(array('parentTagId' => 1, 'keywords' => array('fre-FR' => 'test'), 'mainLanguageCode' => 'fre-FR', 'remoteId' => null, 'alwaysAvailable' => true))
     (vendor/netgen/tagsbundle/bundle/API/Repository/Values/Tags/Tag.php:114)

````
For the correction it is better to correct the typing of the variable in the file bundle/API/Repository/Values/Tags/Tag.php .

````
--- a/bundle/API/Repository/Values/Tags/Tag.php
+++ b/bundle/API/Repository/Values/Tags/Tag.php
@@ -78,7 +78,7 @@ final class Tag extends ValueObject
     /**
      * A global unique ID of the tag.
      */
-    protected string $remoteId;
+    protected ?string $remoteId;
````

Reply:
I already push this correction. https://github.com/netgen/TagsBundle/pull/152/commits/e63608355fd9fd382b1f4670983434cc898842cd

Reply:
Hi. As I said, changing the property type is not a solution, because other parts of code depend on it being only a `string`.

If we still get an error after `'remoteId' => $hashItem['remote_id'] ?? ''` then we need to find why.

Reply:
I'm sorry, I didn't see your change null to ''.

I will try.

Reply:
````
'remoteId' => $hashItem['remote_id'] ?? '' 
````
its working

Reply:
Sorry, finaly it"s not working

````
Argument 'remoteId' is invalid: '' is incorrect value in class 'TagCreateStruct'

Symfony\Component\HttpKernel\Exception\BadRequestHttpException:
Argument 'remoteId' is invalid: '' is incorrect value in class 'TagCreateStruct'

  at vendor/ibexa/core/src/bundle/Core/EventListener/ExceptionListener.php:51
  at Ibexa\Bundle\Core\EventListener\ExceptionListener->onKernelException(object(ExceptionEvent), 'kernel.exception', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ExceptionEvent), 'kernel.exception', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.exception', object(ExceptionEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(ExceptionEvent), 'kernel.exception')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:154)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(ExceptionEvent), 'kernel.exception')
     (vendor/symfony/http-kernel/HttpKernel.php:213)
  at Symfony\Component\HttpKernel\HttpKernel->handleThrowable(object(InvalidArgumentValue), object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:85)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (vendor/autoload_runtime.php:35)
  at require_once('/srv/apps/cms/vendor/autoload_runtime.php')
     (public/index.php:5) 

````